docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,448 @@
+2. Passwords are being stored in code.
+
+We can easily solve these problems because the Snowflake IO manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables, and point Dagster to those environment variables in the run configuration:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v5.py startafter=start endbefore=end
+# repository.py
+
+
+@repository
+def repo():
+    resource_defs = {
+        "local": {
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": {"env": "DEV_SNOWFLAKE_USER"},
+                    "password": {"env": "DEV_SNOWFLAKE_PASSWORD"},
+                    "database": "LOCAL",
+                    "schema": {"env": "DEV_SNOWFLAKE_SCHEMA"},
+                }
+            ),
+        },
+        "production": {
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "system@company.com",
+                    "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS",
+                }
+            ),
+        },
+    }
+    deployment_name = os.getenv("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources(
+            [items, comments, stories], resource_defs=resource_defs[deployment_name]
+        )
+    ]
+```
+
+#### Staging
+
+Depending on your organization’s Dagster setup, there are a couple of options for a staging environment.
+
+For Dagster Cloud users, we recommend using [Branch Deployments](https://docs.dagster.cloud/guides/branch-deployments) as your staging step. A Branch Deployment is a new Dagster deployment that is automatically generated for each git branch. Check out our \[comprehensive guide to Branch Deployments]\(link to claire’s doc) to learn how to use Branch Deployments to verify data pipelines before deploying them to production.
+
+For a self hosted staging deployment, we’ve already done most of the necessary work to run our assets in staging! All we need to do is add another entry to the `resource_defs` dictionary in our repository.
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v5.py startafter=start_staging endbefore=end_staging
+resource_defs = {
+    "local": {...},
+    "production": {...},
+    "staging": {
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": "abc1234.us-east-1",
+                "user": "system@company.com",
+                "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
+                "database": "STAGING",
+                "schema": "HACKER_NEWS",
+            }
+        ),
+    },
+}
+```
+
+### Conclusion
```

Comment:
I know there was some back-and-forth about this, but it feels pretty odd to me to have a `Conclusion` section not be the last section in the guide.

I'd like to push for having the testing section come first. I feel like since it's clearly labeled as optional, users could skip and go to the conclusion if they wanted to.